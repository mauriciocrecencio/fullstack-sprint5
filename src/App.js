import { useEffect, useState } from 'react'
import './App.css'
import Footer from './components/Footer'
import Header from './components/Header/'
import Spinner from './components/Spinner'
import CategoriesContext from './contexts/CategoriesContext'
import FilterContext from './contexts/FilterContext'
import LoadingContext from './contexts/LoadingContext'
import MessageContext from './contexts/MessageContext'
import useLoading from './hooks/useLoading'
import Routes from './routes/Routes'
import CategoriesService from './services/CategoriesService'
import { GlobalStyle } from './styles/GlobalStyle'

function App() {
  const [filter, setFilter] = useState('')
  const [message, setMessage] = useState('')
  const [categories, setCategories] = useState({})
  const [addRequest, removeRequest, isLoading] = useLoading()

  // eslint-disable-next-line
  useEffect(() => loadCategories(), [])

  function loadCategories() {
    addRequest()
    CategoriesService.get()
      .then((c) => setCategories(c))
      .catch(() => setMessage('Ocorreu um erro ao carregar as categorias...'))
      .finally(() => removeRequest())
  }

  return (
    <FilterContext.Provider value={{ filter, setFilter }}>
      <LoadingContext.Provider value={{ addRequest, removeRequest, isLoading }}>
        <MessageContext.Provider value={{ message, setMessage }}>
          <CategoriesContext.Provider value={{ categories }}>
            <Spinner></Spinner>
            <div className="page-container">
              <Header></Header>

              <GlobalStyle />
              <Routes />
            </div>
            <Footer></Footer>
          </CategoriesContext.Provider>
        </MessageContext.Provider>
      </LoadingContext.Provider>
    </FilterContext.Provider>
  )
}

export default App
