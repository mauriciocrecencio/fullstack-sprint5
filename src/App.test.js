/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react'
import App from './App'

// describe('Componente APP principal ', () => {
//   it('deve conter o placeholder do campo de busca de produtos', () => {
//     render(<App />)
//     expect(
//       screen.getByPlaceholderText('O que você está procurando?')
//     ).toBeInTheDocument()
//   })
// })

describe('Componente APP principal ', () => {
  it('deve conter o placeholder do campo de busca de produtos', () => {
    render(<App />)
    expect(
      screen.getByText('Receba ofertas e novidades por e-mail')
    ).toBeInTheDocument()
  })
})

export {}
