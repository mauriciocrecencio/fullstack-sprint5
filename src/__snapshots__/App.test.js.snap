// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente APP principal  deve conter o placeholder do campo de busca de produtos 1`] = `
<input
  class="sc-dIsUp hBRNUh"
  placeholder="O que você está procurando?"
  type="search"
/>
`;

exports[`Componente APP principal  deve conter o placeholder do campo de busca de produtos 2`] = `Promise {}`;

exports[`Componente APP principal  deve conter todos os itens do menu 1`] = `
<input
  class="sc-dIsUp hBRNUh"
  placeholder="O que você está procurando?"
  type="search"
/>
`;
